server/trillian_log_server.go

```diff
@@ -157,6 +181,20 @@ func (t *TrillianLogServer) commitAndLog(tx storage.LogTX, op string) error {
 	return err
 }
 
+func protoToLeaf(proto *trillian.LeafProto) trillian.LogLeaf {
+	return trillian.LogLeaf{Leaf: trillian.Leaf{LeafHash: proto.LeafHash, LeafValue: proto.LeafData, ExtraData: proto.ExtraData}}
+}
+
+func protosToLeaves(protos []*trillian.LeafProto) []trillian.LogLeaf {
+	leaves := make([]trillian.LogLeaf, 0, len(protos))
+
+	for _, proto := range protos {
```

Comment:
I might be missing something, but I don't see how it could end up leaving holes in the result?
But anyway, yeah; I doubt there'd be that much of a difference in performance between the two since as you say you're preallocating, so feel free to leave it as is.